doc/classes/PhysicsShapeQueryParameters3D.xml

```diff
@@ -35,8 +26,19 @@
 		<member name="margin" type="float" setter="set_margin" getter="get_margin" default="0.0">
 			The collision margin for the shape.
 		</member>
+		<member name="shape" type="Resource" setter="set_shape" getter="get_shape">
+			The [Shape3D] that will be used for collision/intersection queries. This stores the actual reference which avoids the shape to be released while being used for queries, so always prefer using this over [member shape_rid].
+		</member>
 		<member name="shape_rid" type="RID" setter="set_shape_rid" getter="get_shape_rid">
-			The queried shape's [RID]. See also [method set_shape].
+			The queried shape's [RID] that will be used for collision/intersection queries. Use this over [member shape] if you want to optimize for performance using the Servers API:
```

Comment:
Looks great! Just one last suggestion (I'm nitpicking now): this code sample doesn't show how to release resources in this case and might mislead users in causing leaks.

Maybe just adding something like this at the end would makes things perfectly clear:
```
# Execute physics queries here...

# Release the shape when done with physics queries.
PhysicsServer3D.free_rid(rid)
```

Reply:
Done, I wonder whether those RIDs are automatically freed by the server when exiting application though, I hope so!

Reply:
Leaks in rid owners are detected on exit and an error is printed:
https://github.com/godotengine/godot/blob/b92477d77e9a6f46e4276a20a451dfac9d485f47/core/rid_owner.h#L289-L299

Which is good I think. In case users forget to free them they are at least notified when the game quits.